test/functional/on.cpp

```diff
@@ -0,0 +1,26 @@
+// Range v3 library
+//
+//  Copyright Eric Niebler 2020
+//
+//  Use, modification and distribution is subject to the
+//  Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at
+//  http://www.boost.org/LICENSE_1_0.txt)
+//
+// Project home: https://github.com/ericniebler/range-v3
+
+#include <range/v3/functional/on.hpp>
+#include <range/v3/functional/concepts.hpp>
+#include "../simple_test.hpp"
+
+using namespace ranges;
+
+int square(int i) { return i * 2; }
```

Comment:
is it intentional to call "times two" "square"?

Reply:
Beat me too it! I was staring at this test forever thinking how does `4` times `16` equal `32`...

Reply:
Yeah I messed up 😂